impl/src/main/java/org/ehcache/impl/persistence/DefaultLocalPersistenceService.java

```diff
@@ -16,74 +16,44 @@
-import org.ehcache.impl.internal.concurrent.ConcurrentHashMap;
-import org.ehcache.spi.persistence.StateRepository;
-import org.ehcache.spi.service.ServiceProvider;
+import org.ehcache.CachePersistenceException;
 import org.ehcache.core.spi.service.FileBasedPersistenceContext;
 import org.ehcache.core.spi.service.LocalPersistenceService;
-import org.ehcache.CachePersistenceException;
+import org.ehcache.impl.config.persistence.DefaultPersistenceConfiguration;
+import org.ehcache.impl.internal.concurrent.ConcurrentHashMap;
 import org.ehcache.spi.service.MaintainableService;
 import org.ehcache.spi.service.Service;
+import org.ehcache.spi.service.ServiceProvider;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.File;
 import java.io.IOException;
 import java.io.RandomAccessFile;
 import java.nio.channels.FileLock;
-import java.nio.charset.Charset;
-import java.security.MessageDigest;
-import java.security.NoSuchAlgorithmException;
-import java.util.ArrayDeque;
-import java.util.Deque;
-import java.util.HashSet;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Set;
-
-import static java.lang.Integer.toHexString;
-import static java.nio.charset.Charset.forName;
-
+import java.util.Map.Entry;
 import java.util.concurrent.ConcurrentMap;
 
-import org.ehcache.config.CacheConfiguration;
+import static org.ehcache.impl.persistence.FileUtils.createLocationIfRequiredAndVerify;
+import static org.ehcache.impl.persistence.FileUtils.recursiveDeleteDirectoryContent;
+import static org.ehcache.impl.persistence.FileUtils.safeIdentifier;
+import static org.ehcache.impl.persistence.FileUtils.tryRecursiveDelete;
+import static org.ehcache.impl.persistence.FileUtils.validateName;
 
 /**
- * Default implementation of the {@link LocalPersistenceService} which can be used explicitly when
- * {@link org.ehcache.PersistentUserManagedCache persistent user managed caches} are desired.
+ * Implements the local persistence service that provides individual sub-spaces for different
+ * services.
  */
 public class DefaultLocalPersistenceService implements LocalPersistenceService {
 
-  private static final Charset UTF8 = forName("UTF8");
-  private static final int DEL = 0x7F;
-  private static final char ESCAPE = '%';
-  private static final Set<Character> ILLEGALS = new HashSet<Character>();
-  static {
-    ILLEGALS.add('/');
-    ILLEGALS.add('\\');
-    ILLEGALS.add('<');
-    ILLEGALS.add('>');
-    ILLEGALS.add(':');
-    ILLEGALS.add('"');
-    ILLEGALS.add('|');
-    ILLEGALS.add('?');
-    ILLEGALS.add('*');
-    ILLEGALS.add('.');
-  }
+  private static final Logger LOGGER = LoggerFactory.getLogger(DefaultLocalPersistenceService.class);
 
-  private final ConcurrentMap<String, PersistenceSpace> knownPersistenceSpaces = new ConcurrentHashMap<String, PersistenceSpace>();
+  private final ConcurrentMap<String, SafeSpace> createdSpaces = new ConcurrentHashMap<String, SafeSpace>();
```

Comment:
Is it really required to maintain this map internally? The user of this service should be maintaining the reference to the `SafeSpaceIdentifier` instances that it prepares? Isn't the `SafeSpaceIdentifier` just a logical representation of the `SafeSpace`? And the `SafeSpaceIdentifier` generation is idempotent using the same owner and identifier values, right? So I'm not sure if we need to maintain it here. Let the caller keep the references to the `SafeSpaceIdentifier` that it has prepared or prepare a new one which will provide the same logical `SafeSpaceIdentifier`. I'm viewing this `SafeSpaceIdentifier` as analogous to `File` which is just a logical representation of a physical file just like `SafeSpaceIdentifier` is for a `SafeSpace`. May be there is something that I'm missing.